internal/dao/dao_service_interface.go

```diff
@@ -0,0 +1,13 @@
+package dao
+
+import (
+	"context"
+)
+
+var GetServiceDao func(ctx context.Context) ServiceDao
+
+// ServiceDao is used for service operations like migrations. All operations are UNSCOPED.
+// See pgx/service_pgx.go for documentation.
+type ServiceDao interface {
+	RecalculatePubkeyFingerprints(ctx context.Context) error
+}
```

Comment:
This is a new special DAO only used in migrations. This allows us to write stub tests if we want to.